src/policy/fees.cpp

```diff
@@ -981,16 +981,17 @@ bool CBlockPolicyEstimator::Read(CAutoFile& filein)
     return true;
 }
 
-void CBlockPolicyEstimator::FlushUnconfirmed(CTxMemPool& pool) {
+void CBlockPolicyEstimator::FlushUnconfirmed() {
     int64_t startclear = GetTimeMicros();
-    std::vector<uint256> txids;
-    pool.queryHashes(txids);
     LOCK(cs_feeEstimator);
-    for (auto& txid : txids) {
-        removeTx(txid, false);
+    size_t num_entries = mapMemPoolTxs.size();
+    // Remove every entry in mapMemPoolTxs
+    while (!mapMemPoolTxs.empty()) {
+        auto mi = mapMemPoolTxs.begin();
+        removeTx(mi->first, false); // this calls erase() on mapMemPoolTxs
```

Comment:
IMO decreasing one by one is avoidable when the end result is to be empty. Prefer something like?
```cpp
LOCK(cs_feeEstimator);
size_t num_entries = mapMemPoolTxs.size();
for (const auto& it = mapMemPoolTxs.begin(); it != mapMemPoolTxs.end(); ++it) {
    const auto& stat = it->second;
    feeStats->removeTx(stat.blockHeight, nBestSeenHeight, stat.bucketIndex, false);
    shortStats->removeTx(stat.blockHeight, nBestSeenHeight, stat.bucketIndex, false);
    longStats->removeTx(stat.blockHeight, nBestSeenHeight, stat.bucketIndex, false);
}
mapMemPoolTxs.clear();
```

Reply:
I didn't want to duplicate the code in `removeTx()` or figure out how to refactor it to be more efficient for this purpose -- since `FlushUnconfirmed()` is only called in shutdown, I don't think the performance savings here from more code changes is worth the review/maintenance burden.

If you'd rather see a bigger refactor here to clean things up, I think I'd prefer to just remove this commit and someone can take it on in a new PR...

Reply:
I think it's fine as it is. I'll take note to measure an alternative implementation.